### PR TITLE
Add generateMaskedInpainting for FLUX.2 base klein

### PR DESCRIPTION
**Pull request by Heisabean**
## Why

FLUX.2-Fill is not yet released by BFL (as of 2026-05). The diffusers
[`FluxFillPipeline`](https://github.com/huggingface/diffusers/blob/main/src/diffusers/pipelines/flux/pipeline_flux_fill.py)
relies on a fine-tuned Fill checkpoint with expanded transformer input
channels (mask + masked-image latents concatenated along channels), so
the channel-cat path does not apply to base klein checkpoints.

Without a Fill checkpoint, the cleanest path on FLUX.2 base is the
**RePaint / SD-inpainting trick** — at each denoising step, the region
outside the user mask is forced back to the original image latents
(re-noised to the next sigma), so only the inside-mask region accumulates
new content.

## What

Adds one public method on `Flux2Pipeline` plus three private helpers,
all in a single extension at the end of `Flux2Pipeline.swift`:

```swift
public func generateMaskedInpainting(
    prompt: String,
    image: CGImage,
    mask: CGImage,           // grayscale, white = inpaint, black = keep
    steps: Int = 4,          // klein distilled default
    guidance: Float = 1.0,
    seed: UInt64? = nil
) async throws -> CGImage
```

Algorithm (Klein distilled, 4 steps):

```swift
for stepIdx in 0..<(sigmas.count - 1) {
    let noisePred = transformer(packedOutputLatents, ...)
    packedOutputLatents = scheduler.step(noisePred, sigma, packedOutputLatents)
    if sigmaNext > 0 {
        let originalNoised = scheduler.scaleNoise(
            sample: imageLatentsPacked,
            sigma: sigmaNext,
            noise: freshNoise)
        packedOutputLatents = (1 - maskPacked) * originalNoised
                             + maskPacked * packedOutputLatents
    }
}
```

Helpers:
- `resolveInpaintDimensions(width:height:)` — clamps to multiple of 32, fits within 1024² area.
- `encodeTextForCurrentModel(prompt:)` — dev/Mistral vs klein/Qwen3 branch.
- `preprocessMaskForLatentBlending(mask:targetHeight:targetWidth:)` — grayscale `CGImage` → `[1, seq, 1]` `MLXArray`, broadcasts naturally against `[1, seq, 128]` latents at blend time. Soft mask values in [0,1] are preserved (caller can pre-binarize if desired).

The extension is in the **same file** so it can access private state (`scheduler`, `transformer`, `vae`, `kleinEncoder`/`textEncoder`, `model`) without exposing any internal API. Existing lines were not modified — only an extension at the file end was added.

## Scope

- Files changed: 1 (`Sources/Flux2Core/Pipeline/Flux2Pipeline.swift`)
- Lines: +200, −0

## When FLUX.2-Fill ships

If/when BFL releases a Fill checkpoint, the higher-fidelity path is
diffusers' channel-cat conditioning. This RePaint method can be kept as
a fallback for base klein (e.g., when users don't want the Fill model's
disk/memory footprint), or removed entirely.

## Tested

- Code compiles (`swift build --target Flux2Core`, 48s).
- Downstream consumer (Pawworks SnapPaw screenshot redactor) wires the
  method via a thin actor and Settings UI in macOS 15.

Happy to adjust signatures, naming, or split into a generation-mode
enum (`Flux2GenerationMode.maskedInpainting`) if preferred — let me know.

### CODE REVIEW

**Comment by VincentGourbin** [comment]
Hi @Heisabean — superseded by #85, which keeps your RePaint algorithm but re-integrates it as a `Flux2MaskedInpaintingChain` over a new `Flux2StepHook` extension point on `Flux2Pipeline.generate*`. The chain inherits progress callbacks / profiler / memory caps from the standard path, and the guidance tensor is now built via `model.usesGuidanceEmbeds` (your version passed `MLXArray([guidance])` unconditionally, which trips on Klein distilled). The same primitive composes into `Flux2OutpaintingChain` for free. Thanks for the seed PR — credited in the chain source. Closing this one as superseded.
